src/features/linters/ruff.ts

```diff
@@ -3,18 +3,31 @@ import { CancellationToken, OutputChannel, TextDocument, Uri, workspace } from '
 import { ILinterInfo, ILintMessage, LintMessageSeverity } from '../../types';
 import { BaseLinter } from './baseLinter';
 
-const REGEX = '(?<file>.*.py):(?<line>\\d+):(?<column>\\d+): (?<code>[\\w-]+) (?<message>.*)\\r?(\\n|$)';
 const COLUMN_OFF_SET = 1;
 
+interface IRuffLocation {
+  row: number;
+  column: number;
+}
+
+interface IRuffLintMessage {
+  kind: string | { [key: string]: any[] };
+  code: string;
+  message: string;
+  location: IRuffLocation;
+  end_location: IRuffLocation;
+  filename: string;
+}
+
 export class Ruff extends BaseLinter {
   constructor(info: ILinterInfo, outputChannel: OutputChannel) {
     super(info, outputChannel, COLUMN_OFF_SET);
   }
 
-  protected async runLinter(document: TextDocument, cancellation: CancellationToken): Promise<ILintMessage[]> {
+  protected async runLinter(document: TextDocument, _cancellation: CancellationToken): Promise<ILintMessage[]> {
```

Comment:
Unsure what to do here as well. I thought ESLint(?) would ignore unused arguments prefixed with `_`.